### PR TITLE
Fix: timeline doesnt try to parse non-existent datetime

### PR DESCRIPTION
**Pull request by phlax**
This is a fix for #4805, altho i think this code needs to be
refactored, and tests obviously need some improvement.

### CODE REVIEW

**Comment by unho** [comment]
lgtm

File: pootle/apps/pootle_store/views.py [hunk 1]
```diff
@@ -380,11 +380,16 @@ def render_timeline(self, context):
     def get_entries_group_data(self, context):
         result = []
         for entry_group in context['entries_group']:
+            display_dt = entry_group['datetime']
+            if display_dt is not None:
+                display_dt = dateformat.format(display_dt)
+                iso_dt = entry_group['datetime'].isoformat()
+            else:
+                iso_dt = None
```
Comment (ta2-1) [inline pootle/apps/pootle_store/views.py]:
```python
iso_dt = None
if display_dt is not None:
    display_dt = dateformat.format(display_dt)
    iso_dt = entry_group['datetime'].isoformat()
```
won't decrease coverage )

**Comment by ta2-1** [comment]
lgtm

File: pootle/apps/pootle_store/views.py (hunk 1 above)
Comment (phlax) [inline pootle/apps/pootle_store/views.py]: it should decrease coverage - i did it this way intentionally

**Pull request merged**